### DISCUSSION TITLE
[] is now used for generics instead of .<>             Why???????

### DISCUSSION THREAD

**Discussion by Jengro777** [vlang/v#17194] [General]
<> use generics is better, I think.

**Comment by JalonSolov** [comment]
From discord...

Alex M
 — 
11/27/2022 10:03 AM
Like Scala and Go, V recently switched to using [] for generics.

Using <> resulted in parsing issues that couldn't be resolved without type information.

Old code will keep working, vfmt automatically takes care of the change.

You can read about the benefits of using [] here:

https://soc.me/languages/stop-using-angle-brackets-for-generics.html

https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#why-not-use-the-syntax-like-c_and-java
